### ISSUE TITLE
Script > Settings does absolutely nothing with the attempts at embedded default fonts.

### ISSUE THREAD

**Issue by ghost** [trelby/trelby#34]
![trelby](https://github.com/user-attachments/assets/a462763e-39c4-4599-936f-f90623d7f966)
I cannot at all change the settings using the GUI. It always gives me this error every setting I try to change. Going through the source code, it seems you're trying to embed the font yet it doesn't use the default font. I managed to manually set the font using [courier prime](https://quoteunquoteapps.com/courierprime/) which is under the [OFL license](https://openfontlicense.org/). I really think courier prime should just be bundled at this point because it's completely broken. It didn't do anything when I clicked number scenes, but that's a different issue for a different day.
[courier-prime.zip](https://github.com/user-attachments/files/17085385/courier-prime.zip)

**Comment by rcentros** [comment]
Please don't change the way Trelby works with PDF fonts.

The way Trelby works with PDF fonts (not display fonts) since 2012 is by pointing to the actual font files in a specific directory. In my case I have my Trelby fonts in this directory... ~/Documents/scripts/fonts. Once there you can browse for whatever font you want to use in your PDF document. This allows you to set up custom script settings and save them to an .sconf file. (You can even edit these .sconf files by hand.) For example, I can creat a LetterCourierScreenplay option (using the Letter paper size) the Courier Screenplay font or a LetterCourierPrime option with the Prime font (or whatever other font I may want to use) and then just Load it in.

Or, I can do it on the fly, as shown in this (lousy) video. (I have trouble making these.)

https://drive.google.com/file/d/10j0GkAqSfcwKZOhPz1zr7i5RVlj6gRlT/view

I think a better solution than changing this feature (or limiting Trelby to one font) is to simply let people know how Trelby works for PDF fonts. For me, taking away this ability to customize would be a big step backwards.

**Comment by priframr** [comment]
@rcentros 
I won't be making any radical changes on how trelby works with fonts. However, it was a bug this entire time. The line in question was:
`if pf.name and not pf.filename:`
The damning part of this was pf.name, it was always going to be set no matter what because it was going to use one of the four categories for the name. I changed it to pf.pdfName and that was the actual font name it was looking for. This was still a bug in the end and I believe I fixed it.

**Comment by rcentros** [comment]
I'm sorry I overreacted. I saw something about Courier Prime and misread that it was going to be the only font available. (Though it probably would make a good default font, as a lot of people like it.) 

I've used Trelby since it came out in 2012. I was a beta tester for Blyte in 2004 and even suggested to Osku that he name it something other than Blyte when he released version 1.0 in December of that year. (Oddly enough one of my name suggestions was Fade In (the other was Smash Cut). I think naming it Trelby in 2012 (instead of Trilby) was genius as an Internet search for Trelby only finds this application.  A few years ago I tried to learn Python with the aim of updating Trelby for newer versions of Linux (but I'm definitely not a programmer and that was a failure).

All this said to say that I was very happy to see new versions of Trelby starting to come out a few years back -- and now seeing it come to FlatHub now means I can just suggest people use it and not worry about explaining what version to get and how to install it for their distribution (though I'm happy to see it in the Fedora repository and I like that a .deb version is available at Suse).

Sorry to ramble. I was wondering if it would be possible to start a Trelby forum on Reddit (or somewhere else) so that idiots like me, asking stupid questions like this, wouldn't "gunk up" the "Issue" forum of GitHub. For example, I know that there was talk of supporting Unicode in Trelby, but I don't know if that was ever implemented (that question was asked by a friend last night). I also see mention of MacOS installs and it would be nice to know how to do that so I could tell Mac users how to install Trelby if they wanted it. (I have an old Mac basically for testing things like this.)

At any rate, I thank you for all your work and would like to help promote Trelby in some way. Maybe some kind of forum would help with that.

**Comment by limburgher** [comment]
The main issue with bundling fonts is that some Linux distributions forbid it. What I have in mind going forward is to make better use of the font management functions in wxPython once 4.1 is more ubiquitous.

**Comment by makrbrk** [comment]
> The main issue with bundling fonts is that some Linux distributions forbid it. What I have in mind going forward is to make better use of the font management functions in wxPython once 4.1 is more ubiquitous.

What fonts are you referring to and what distros forbid it? I'm curious to hear now what you meant by that.

**Comment by limburgher** [comment]
Fedora is a good example: [https://docs.fedoraproject.org/en-US/packaging-guidelines/FontsPolicy/](https://docs.fedoraproject.org/en-US/packaging-guidelines/FontsPolicy/)

**Comment by makrbrk** [comment]
> Fedora is a good example: https://docs.fedoraproject.org/en-US/packaging-guidelines/FontsPolicy/

OFL is listed as a allowed font license.
https://fedoraproject.org/wiki/Licensing:Fonts

**Comment by limburgher** [comment]
True. But if we shipped this font in Trelby I'd have to remove it and symlink it to the version shipped in Fedora.

**Comment by anilgulecha** [comment]
> I was wondering if it would be possible to start a Trelby forum on Reddit

You should do it @rcentros - I see that you've been a user for a while, and like it for what it is. Start the community, and I think we'd be happy to point to it from the website etc. Do set it up as a friendly, supportive place - I've seen some subreddits be toxic.

**Comment by rcentros** [comment]
@anilgulecha 
I've created a Trelby subReddit (community). As I learn more I hope to make it look and work a little nicer, but I'm mostly interested in the content. This will be mostly for users, like me, but I'm pretty sure issues will come up (if anyone finds the subReddit) so they'll probably passed along to to here.

The subReddit's URL is https://www.reddit.com/r/Trelby/

**Comment by janopae** [comment]
It seems like this issue is not about changing fonts, but instead, you experience this message every time you try to change any setting, @ghost?

You can use 2 kinds of fonts Trelby:
1. PDF standard fonts
2. custom fonts

Custom fonts need a font file path, and they will be embedded in PDF files to be exported from Trelby. You seem to work around this issue now by using a custom font.

It should, however, also be possible to set one of [the 14 PDF standard fonts](https://kbpdfstudio.qoppa.com/standard-14-pdf-fonts/): These fonts are part of the PDF standard and are *required* to be available on any system with a PDF reader. They only need to be referenced by name, and they don't need to be bundled with the PDF file. Therefore, we don't need to bundle them with Trelby. That's why trelby uses one of these standard fonts by default.

What happens when you leave both, the name and the path field, blank? Does it save the settings then?

Maybe the check that spawns the message isn't smart enough to detect the standard fonts and just assumes use either use the default (leave both fields blank) or specify a font file.

**Comment by limburgher** [comment]
Fixed in master, coming in 2.4.14
